workers/unity/Packages/io.improbable.gdk.gameobjectcreation/EntityTypeExpectations.cs

```diff
@@ -8,29 +10,34 @@ namespace Improbable.Gdk.GameObjectCreation
     /// </summary>
     public class EntityTypeExpectations
     {
-        private Type[] defaultExpectation;
+        private ComponentType[] defaultExpectation;
 
-        private readonly Dictionary<string, Type[]> entityExpectations
-            = new Dictionary<string, Type[]>();
+        private readonly Dictionary<string, ComponentType[]> entityExpectations
+            = new Dictionary<string, ComponentType[]>();
 
-        public void RegisterDefault(Type[] defaultComponentTypes)
+        public void RegisterDefault(Type[] defaultComponentTypes = null)
         {
-            defaultExpectation = defaultComponentTypes;
+            defaultExpectation = defaultComponentTypes?
+                .Select(type => new ComponentType(type, ComponentType.AccessMode.ReadOnly))
+                .ToArray();
```

Comment:
This should never be null. If the user gives `null` it should be an empty array.

Reply:
(Note on L37 we just return `defaultExpectation`)

Reply:
Also its odd to have a single optional param. Would it make more sense for `defaultExpectation` to be an empty array by default and then you call with method (always with a valid type array)  to override it?

Reply:
> Would it make more sense for defaultExpectation to be an empty array by default and then you call with method (always with a valid type array) to override it?

That is the behaviour we have now (latest commit)

defaultlist: empty
RegisterDefault(): resets defaultlist to empty
RegisterDefault(null): resets defaultlist to empty
RegisterDefault(yourlist): sets defaultlist to yourlist